storage/storage.go

```diff
@@ -345,13 +346,39 @@ func constructWhereClauseForContent(reportRules types.ReportRules) string {
 	return statement
 }
 
+func getExtraDataFromReportRules(rules []types.RuleContentResponse, reportRules types.ReportRules) []types.RuleContentResponse {
+	if len(reportRules.HitRules) == 0 {
+		return rules
+	}
+
+	for _, ruleContent := range rules {
+		module := ruleContent.RuleModule
+		//errorKey := ruleContent.ErrorKey
+
+		for _, hitRule := range reportRules.HitRules {
+			moduleOnReport := strings.TrimSuffix(hitRule.Module, ".report")
+			if module == moduleOnReport {
+				templateData, err := json.Marshal(hitRule.Details)
+				if err != nil {
+					log.Error().Err(err).Msg("Error while marshalling rule details from the report")
+					continue
+				}
+				ruleContent.TemplateData = string(templateData)
```

Comment:
@natiiix @tisnik @Sergey1011010 guys im having trouble here, I might be too dumb today to see it, the `templateData` is correct according to debugging but its just always empty when returned from API.

Reply:
Because you're changing the copy https://play.golang.org/p/_pLy40Kstwh of a value

Reply:
I'd change to something like this:
```
for i := range array {
    item := &array[i]
}
```

Reply:
thanks man.... :1st_place_medal: